src/main/java/core/basesyntax/RobotRoute.java

```diff
@@ -2,6 +2,47 @@
 
 public class RobotRoute {
     public void moveRobot(Robot robot, int toX, int toY) {
-      //write your solution here
+        RobotRoute.routeForx(robot, toX);
+        RobotRoute.routeFory(robot, toY);
+    }
+
+    private static void routeForx(Robot robot, int toX) {
+        int x = robot.getX();
+        if (x > toX) {
+            while (robot.getDirection() != Direction.LEFT) {
+                robot.turnLeft();
+            }
+            while (robot.getX() != toX) {
+                robot.stepForward();
+            }
+        }
+        if (x < toX) {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnRight();
+            }
+            while (robot.getX() != toX) {
+                robot.stepForward();
+            }
+        }
+    }
+
+    private static void routeFory(Robot robot, int toY) {
+        int y = robot.getY();
+        if (y < toY) {
+            while (robot.getDirection() != Direction.UP) {
+                robot.turnLeft();
+            }
+            while (robot.getY() != toY) {
+                robot.stepForward();
+            }
+        }
+        if (y > toY) {
```

Comment:
Use 'else if' instead of 'if' to avoid unnecessary condition checks when the previous condition is true.